src/battle_util.c

```diff
@@ -7517,8 +7517,13 @@ bool32 IsBattlerProtected(u8 battlerId, u16 move)
         else if (gProtectStructs[battlerId].protected)
             return FALSE;
     }
-    
-    if (!(gBattleMoves[move].flags & FLAG_PROTECT_AFFECTED))
+
+    // Protective Pads doesn't stop Unseen Fist from bypassing Protect effects, so IsMoveMakingContact() isn't used here.
+    // This means extra logic is needed to handle Shell Side Arm.
+    if (GetBattlerAbility(gBattlerAttacker == ABILITY_UNSEEN_FIST)
```

Comment:
`if (GetBattlerAbility(gBattlerAttacker == ABILITY_UNSEEN_FIST)` should be `if (GetBattlerAbility(gBattlerAttacker) == ABILITY_UNSEEN_FIST)`